src/Report_Query.php

```diff
@@ -614,6 +623,97 @@ protected function prime_caches( $result_ids ) {
 		}
 	}
 
+	/**
+	 * Adds any results from the given IDs to the log data cache that do not already exist in cache.
+	 *
+	 * @since 0.1.1
+	 *
+	 * @global wpdb $wpdb WordPress database abstraction object.
+	 *
+	 * @param array $result_ids List of result IDs.
+	 */
+	protected function prime_log_data_caches( $result_ids ) {
+		global $wpdb;
+
+		$non_cached_ids = array();
+		foreach ( $result_ids as $result_id ) {
+			if ( ! wp_cache_get( (string) $result_id . '_log_data', Reports::CACHE_GROUP ) ) {
+				$non_cached_ids[] = $result_id;
+			}
+		}
+		if ( empty( $non_cached_ids ) ) {
+			return;
+		}
+
+		$table_name = Plugin::instance()->report_logs()->get_db_table_name();
+
+		// Request regular columns for each report.
+		$fresh_results = $wpdb->get_results( // phpcs:ignore WordPress.DB.DirectDatabaseQuery
+			sprintf(
+				// phpcs:ignore WordPress.DB.PreparedSQL
+				"SELECT report_id, COUNT(*) AS count, MIN(triggered) AS first_triggered, MAX(triggered) AS last_triggered, MIN(reported) AS first_reported, MAX(reported) AS last_reported FROM {$table_name} WHERE report_id IN (%s) GROUP BY report_id",
```

Comment:
This could still use prepared SQL, like:

```php
$ids_in_sql = implode( ',', array_fill( '%d', count( $non_cached_ids ) ) );
$non_cached_ids = array_map( 'intval', $non_cached_ids ); // For good measure.
$fresh_results = $wpdb->get_results(
    $wpdb->prepare( "SELECT report_id, COUNT(*) AS count, MIN(triggered) AS first_triggered, MAX(triggered) AS last_triggered, MIN(reported) AS first_reported, MAX(reported) AS last_reported FROM {$table_name} WHERE report_id IN ($ids_in_sql) GROUP BY report_id", $non_cached_ids ),
    ARRAY_A
);
```

Also below.

Reply:
@westonruter I implemented this in the latest commit.